### ISSUE TITLE
Option to remove tests and comments

### ISSUE THREAD

**Issue by tarriel** [jfaixo/cargo-merge#1]
After working on my submissions for a while I find I tend to run out of space.
This is because I like comments in my code so later I can remember what the heck I was trying to achieve.
I also have `#[test]`s in my code, my old workflow usually has
`#[cfg(test)] 
mod tests;`
at the bottom.  this compiles even without the tests.rs file.  Which lets me keep my tests in a separate file and they aren't part of my submission.
Having tests lets me debug my code locally, and I can run the same thing as many times as required to understand how stupid I am.  It also lets me stop regressions, where I fixed a corner case, I can be reminded about it again, when a change elsewhere causes it to fail.
Cargo merge kindly adds the tests back into my submission.
It would be nice to have an option to remove tests, remove comments, remove allow()s, and maybe minify my code by removing the indenting.  It is harder to understand errors if all line breaks are removed, but there are a lot of spaces which can be removed while keeping line breaks, besides Rust Analyser is pretty good at getting code which just works, so it is usually only my own asserts and unwraps that cause the errors.
As an example, one of my files had 610 lines, (probably 450 lines without comments) after using cargo merge, I now have 2155 lines.
Something like the -s for removing eprint statements would be nice.

**Comment by tarriel** [comment]
As requested, from gitlab.  The comments definitely cause a problem at codingame, I just tested it as I was sure I had that problem before.  Perhaps they allow more when it is comments?
`An error occurred (#314): "Submitted code is too big".`
I'm not sure how to tag this with a label like enhancement.

**Comment by jfaixo** [comment]
Yeah sorry I have not defined any way of interacting with this tiny repository yet, i'll try to organize a little bit, and keep in touch with you in this thread :)

**Comment by jfaixo** [comment]
Having tests in a separate file is pretty neat :+1: I usually have directly related tests right below in a `#[cfg(test)] mod tests { ... }` scope...

I'll try to handle both.

I have created issues to track the 3 features: #2 #3 #4
